src/Microsoft.Graph.Core/Requests/Content/BatchResponseContent.cs

```diff
@@ -77,6 +77,58 @@ public async Task<HttpResponseMessage> GetResponseByIdAsync(string requestId)
             return GetResponseMessageFromJObject(jResponseItem);
         }
 
+        /// <summary>
+        /// Gets a batch response as a requested type for the specified batch request id.
+        /// </summary>
+        /// <param name="requestId">A batch request id.</param>
+        /// <returns>A deserialized object of type T<see cref="HttpResponseMessage"/>.</returns>
+        public async Task<T> GetResponseByIdAsync<T>(string requestId)
+        {
+            using (var httpResponseMessage = await GetResponseByIdAsync(requestId))
+            {
+                var responseHandler = new ResponseHandler(new Serializer());
+                
+                if (!httpResponseMessage.IsSuccessStatusCode)
```

Comment:
We need to discuss why this isn't being done inside the response handler.  The code you have written is good, but we shouldn't need to duplicate this code here in BatchResponseContent.  We can fix this later without a breaking change.